File: solvers/toy/coefs_toy.m
function C = coefs_toy(data)

    A = data{1};

    C = zeros(3,10);

    C([3,6,9,11,13,17,19,23,25]) = 1;
    C(28) = A(1);
    C(29) = A(2);
    C(30) = A(3);

    C = C./repmat(sqrt(sum(C.*conj(C),2)),1,size(C,2));

end

File: solvers.python/py_toy/coefs_toy.py
import numpy as np

def coefs_toy(data):
    A = data[0].T.flat

    C = np.zeros((3,10))
    C.flat[[4,6,8,13,15,17,20,21,22]] = 1
    C.flat[9] = A[0]
    C.flat[19] = A[1]
    C.flat[29] = A[2]

    C = C/np.sqrt(np.sum(C*np.conj(C),axis=1,keepdims=True))

    def U(x,y,z):
        return np.array([x**3,y**3,z**2*x,x**2,x*y,y**2,y*z,z**2,z,1])

    return C,U